feat: basic roommates page frontend

aidandenlinger opened the pull request:
Closes #14.

User Story: I want to browse through a list of recommended roommates so I can find a roommate that matches my interests.

![Screenshot 2023-02-26 at 03-35-03 Roommate Suggestions](https://user-images.githubusercontent.com/47428697/221391085-4c4c30d7-9ad6-4d16-84e1-9e00572838f0.png)

This PR creates a basic layout to display roommates. It does *not* contain the filter component, which is in #48. We can either wait for #48 to get merged, integrate it here and then merge here - or merge both and do a separate PR to combine them, either way. It contains a `RoommateCard` component that can take a Roommate and turn it into a card as shown. Notably, this does not implement the design from Figma:

![Screenshot from 2023-02-25 19-35-35](https://user-images.githubusercontent.com/47428697/221391127-810b8352-8a1d-4a74-8d6b-881b2a39a690.png)

I prioritized being similar to the Properties page and implementing this as a Card. The Figma design implements this page as a set of mini-profile pages, but I feel this would look distractedly different from the Properties page. We can discuss this in Sunday's meeting, I'm happy to change the card to look exactly like this if we want to take that route.

This PR also sets the root of our website to go immediately to the roommates page.

Requesting a review from Aditya since he did the properties page to see if he thinks that these pages are similar or not.

aidandenlinger commented on src/components/RoommateGrid/RoommateCard.js:
```diff
@@ -0,0 +1,40 @@
+export default function RoommateCard({ user }) {
+	// TODO: Default profile picture if we don't have an imageUrl
+	return (
+		<Card variant="outlined" sx={{width: 500}}>
```
`width:500` is a bit nonsensical, I just wanted the Card to take up as much width as it could. Let me know if there's a better way.

jn1118 replied:
Could also try Fullwidth or set the position and width as this: https://stackoverflow.com/a/44550716 ?

aidandenlinger commented:
Per meeting review: we should use `Avatar` instead of stretching the entire profile pic, will look weird. It's worth diverting from the look of the Property cards

lmy0125 commented on next.config.js:
```diff
@@ -3,7 +3,18 @@ const nextConfig = {
   typescript: {
     ignoreBuildErrors: true,
-  }
+  },
+  async redirects() {
+    return [
+      {
+        // Redirect / (root of our website) to /roommates
+        // https://nextjs.org/docs/api-reference/next.config.js/redirects
+        source: '/',
+        destination: '/roommates',
+        permanent: false,
+      },
+    ]
+  },
```
I think we can omit this redirect for now, because we need some way to view/utilize the index.js file somehow. If we redirect out from the root, index.js becomes useless. We can discuss routing in meeting.

aidandenlinger replied:
sounds good. i did this redirect because I thought the roommate page *was* the front page and index.js is intended to be useless - if we don't want it to be we should definitely discuss

lmy0125 reviewed (changes requested):
Also need to pull from main since I just merged Sia's code.

aidandenlinger commented:
Todo list:
- change to show Avatar
- add Sia's filtering component
- remove routing

aidandenlinger commented:
Notes:
- Implemented Avatar for the profile picture instead of card background image. The old version had 491.333 px * 140 px = 68786.62 px^2 of duck image per roommate card, while the Avatar version has pi * 36^2 = 4071.504 px^2 of duck image per roommate card. This is almost 17 times less duck.

- I introduced @jn1118's Roommate filtering menu. However, it has issues with overflowing when the screen size gets small - not sure if this is something I need to change in my code or hers. I'm just putting it in a grid in roommates.js.
  - ![Screenshot 2023-03-01 at 02-07-40 Roommate Suggestions](https://user-images.githubusercontent.com/47428697/222026465-e5b224d9-a505-4a94-a93c-2e7d80ae5532.png)
  - ![Screenshot 2023-03-01 at 02-08-05 Roommate Suggestions](https://user-images.githubusercontent.com/47428697/222026480-ff297a15-a337-42b8-a4ee-3fc172e3ea20.png)

- Removed routing. However, I think that we should introduce routing *tonight* - otherwise we load up our website for Satvik tomorrow, and it immediately shows the index.js page which doesn't really have any content. I don't particularly care where we route to, but I think anything is better than that for now.

lmy0125 commented:
I will add the routing on my PR later then.

lmy0125 commented:
Check comment on Vercel. Is the double scrolling bars because of the introduction of the filter component?

aidandenlinger commented:
it's not the filter element, I can get rid of it completely and i'm still getting double scrolling. super, super weird, digging into it now

lmy0125 commented:
Maybe you can checkout my PR since that has a similar grid layout.

aidandenlinger commented:
ah okay found it! Weirdly enough it was caused by

```
-    <Grid container columns={{ xs: 1, sm: 2, md: 3 }} spacing={10} margin={5}>
+    <Grid container columns={{ xs: 1, sm: 2, md: 3 }} spacing={10}>
```

just by removing the margin on my grid it caused the double scroll? weird but fixing

aidandenlinger commented:
Okay got it - if your page is just a grid, that grid needs a margin or else it gets angry and messes with your scrolling >:(

lmy0125 reviewed (approved):
LGTM
